symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV2.java

```diff
@@ -0,0 +1,185 @@
+package com.symphony.bdk.core.service.datafeed.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.AckId;
+import com.symphony.bdk.gen.api.model.V4Event;
+import com.symphony.bdk.gen.api.model.V5Datafeed;
+import com.symphony.bdk.gen.api.model.V5EventList;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A class for implementing the datafeed v1 service.
+ */
+@Slf4j
+public class DatafeedServiceV2 extends AbstractDatafeedService {
+
+    private final AtomicBoolean started = new AtomicBoolean();
+    private final AckId ackId;
+    private V5Datafeed datafeed;
+
+    public DatafeedServiceV2(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, podClient, authSession, config);
+        this.ackId = new AckId().ackId("");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws ApiException, AuthUnauthorizedException {
+        if (this.started.get()) {
+            throw new IllegalStateException("The datafeed service is already started");
+        }
+        try {
+            this.datafeed = this.retrieveDatafeed();
+            if (this.datafeed == null) {
+                this.datafeed = this.createDatafeed();
+            }
+            log.debug("Start reading datafeed events");
+            do {
+                this.started.set(true);
+                this.readDatafeed();
+            } while (this.started.get());
+        } catch (Throwable e) {
```

Comment:
it possible to improve this exception catch?

Reply:
because the `Retry#executeCheckedSupplier` throws a `Throwable` then this block of code have to catch `Throwable`.

Reply:
we could have specific catches for each type and then at the end a catch for Throwable
(SonarLint is reporting this as a warning in IDEA, I highly recommend installing this plugin ;) it helps to catch Sonar problems before having it reported in the PR, and it is also reporting more issues given how old the Sonar PR checker is....)